capstone/fitness/src/main/java/com/google/sps/servlets/AbstractAuthServlet.java

```diff
@@ -1,42 +1,153 @@
 public class AbstractAuthServlet extends AbstractAppEngineAuthorizationCodeServlet {
-  String nickname = UserServiceFactory.getUserService().getCurrentUser().getNickname();
 
+  String APPLICATION_NAME = "Marathon App";
+  String colorId = "4";
+  Calendar calendar;
+  Scheduler scheduler;
+  long exerciseDuration = 30;  
+  Gson gson = new Gson();
+  DateTimeFormatter dtf = DateTimeFormatter.ofPattern("MM/dd/YYYY");  
+  LocalDateTime now = LocalDateTime.now();  
+
+  
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    System.out.println("Nickname " + nickname);
-    response.setContentType("text/html;");
-    response.getWriter().println(nickname + " has authorized this app.");
-  }
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
+    response.setContentType("application/json;");
+    
+    // Build calendar. 
+    String userId = getUserId(request);
+    Credential credential = Utils.newFlow().loadCredential(userId);
+    calendar = new Calendar.Builder(new UrlFetchTransport(), new JacksonFactory(), credential).setApplicationName(APPLICATION_NAME).build();
+
+    scheduler = new Scheduler(exerciseDuration);
+    
+    String wks = (DataHandler.getData("weeksToTrain",DataHandler.getUser()));
+    int weeksToTrain = Integer.parseInt(wks);
+    
+    int daysAvailable = weeksToTrain * Time.weeksToDays;
+    
+    List<String> exercises = this.getExercises();
+    int timesPerWeek = daysAvailable/ exercises.size(); 
 
+    // // Sets minSpan to 7:00 AM the next day, and maxSpan to 7PM the next day.
+    LocalDateTime now = LocalDateTime.now();  
+    DateTimeFormatter myDtf = DateTimeFormatter.ofPattern("YYYY-MM-dd");  
+    String nextDayOfMonth = myDtf.format(now.plusDays(1));
+    String nextDayStartTime = "T11:00:00+00:00";
+    String nextDayEndTime = "T23:00:00+00:00";
+    String minRCF3339 = nextDayOfMonth + nextDayStartTime;
+    String maxRCF3399 = nextDayOfMonth + nextDayEndTime;
+    DateTime minSpan = new DateTime(minRCF3339);
+    DateTime maxSpan = new DateTime(maxRCF3399);
+
+    // Use the scheduler to schedule one exercise per day. 
+    int y = 0;
+    for (int x = 0; x<daysAvailable; x+=timesPerWeek){
+        if (y>=exercises.size()){
+          break;
+        }
+        else{
+          List<Event> currentlyScheduledEvents = this.getEventsInTimespan(minSpan, maxSpan);
+          Event exerciseEvent = scheduler.getFreeTime(minSpan, maxSpan, currentlyScheduledEvents);
+          exerciseEvent.setSummary(APPLICATION_NAME + ": " + exercises.get(y));
+          exerciseEvent.setColorId(colorId);
+          // TODO (@piercedw) : Store each event's eventID in datastore for display later.
+          this.insertEvent(exerciseEvent);
+          // Increment minSpan and maxSpan by one day. 
+          minSpan = new DateTime(minSpan.getValue() + (Time.millisecondsPerDay * timesPerWeek));
```

Comment:
Consider using datetime api for time calculation, https://www.baeldung.com/java-8-date-time-intro